GVFS/GVFS.Platform.Mac/MacFileSystemVirtualizer.cs

```diff
@@ -560,31 +559,19 @@ private void OnFileRenamed(string relativeDestinationPath, bool isDirectory)
         {
             try
             {
-                Result result;
-                try
-                {
-                    IEnumerable<ProjectedFileInfo> projectedItems;
-
-                    // TODO: Pool these connections or schedule this work to run asynchronously using TryScheduleFileOrNetworkRequest
-                    using (BlobSizes.BlobSizesConnection blobSizesConnection = this.FileSystemCallbacks.BlobSizes.CreateConnection())
-                    {
-                        projectedItems = this.FileSystemCallbacks.GitIndexProjection.GetProjectedItems(CancellationToken.None, blobSizesConnection, relativePath);
-                    }
+                IEnumerable<ProjectedFileInfo> projectedItems;
 
-                    result = this.CreatePlaceholders(relativePath, projectedItems, triggeringProcessName);
-                }
-                catch (SizesUnavailableException e)
+                // TODO: Pool these connections or schedule this work to run asynchronously using TryScheduleFileOrNetworkRequest
+                using (BlobSizes.BlobSizesConnection blobSizesConnection = this.FileSystemCallbacks.BlobSizes.CreateConnection())

```

Comment:
Since this isn't even used when `populateSizes` is `false`, should we pass `null` to `GetProjectionItems` and remove this code?

Reply:
And maybe that can be the indicator to get the sizes is if the `blobSizesConnection` is not null instead of adding the `populateSizes` parameter?

Reply:
Both great suggestions, thanks!